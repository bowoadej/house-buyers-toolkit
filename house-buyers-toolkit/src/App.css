#root {
  max-width: 100%;
  margin: 0 auto;
  text-align: center;
}

body {
  max-width: 100%;
}

.header {
  position: top;
  background-color: blue;
  color: lightcyan;
  width: 100vw;
}

.viewing-checklist {
  display: grid;
}

.viewingChecklist {
  position: sticky;
}

.navigation-bar {
  background-color: gray;
  margin-top: -1.5%;
  margin-bottom: -1.6%;
}

ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

li {
  display: inline;
  padding-right: 1%;

}

.monthly-repayment-calculator {
  margin-top: -1.5%;
}

.accordion-monthly-repayment-calculator {
  outline: 5px solid lightblue;
  display: inline-flex;
}

.accordion-item-viewing-checklist {
  outline: 5px solid lightblue;
  display: inline-flex;
}

.accordion-item-maximum-borrowing-amount {
  outline: 5px solid lightblue;
  display: inline-flex;
}