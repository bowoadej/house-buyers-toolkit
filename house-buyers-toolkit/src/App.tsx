import './App.css'
import "./fonts/Urbanist-Bold.woff2"
import "./fonts/Urbanist-Regular.woff2"
import "./fonts/Urbanist-SemiBold.woff2"
import ViewingChecklist from './components/ViewingChecklist'
import Navigation from './components/Navigation'
import MonthlyRepaymentCalculator from './components/MonthlyRepaymentCalculator'
import MaximumBorrowingAmount from './components/MaximumBorrowingAmount'
import Disclaimer from './components/Disclaimer'
import { FeatureFlagsContext } from "../src/components/FeatureFlags";
import { useContext } from 'react'
import LoginForm from './components/LoginForm'
import Header from './components/Header'

function App() {

  const checklist = useContext(FeatureFlagsContext).renderViewingChecklist;
  const calculator = useContext(FeatureFlagsContext).renderMonthlyRepaymentCalculator;
  const maximumborrowingamount = useContext(FeatureFlagsContext).renderMaximumBorrowingAmount;
  const navigation = useContext(FeatureFlagsContext).renderNavigationBar;

  return (
    <>
      <div className='body'>
<<<<<<< Updated upstream
<<<<<<< Updated upstream
=======
=======
>>>>>>> Stashed changes
  {
    login &&
      <Topline />
  }
>>>>>>> Stashed changes
        <Header />
        <div className='content'>
          {
            navigation &&
            <Navigation />
          }
          {checklist &&
            <ViewingChecklist />
          }
          {maximumborrowingamount &&
            <MaximumBorrowingAmount />
          }
          {calculator &&
            <MonthlyRepaymentCalculator />
          }
          <br></br>
          <div className='footer'>A Toolkit Created By <a href='https://bowoadej.com'>Bowo Adejuyigbe.</a>Copyright 2025. All rights reserved</div>
          <br></br>
          <Disclaimer />
        </div>
      </div >
    </>
  )
}

export default App 