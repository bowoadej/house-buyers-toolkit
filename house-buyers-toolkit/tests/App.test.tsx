//import React from 'react';
import { /*describe,*/ expect,/* jest,*/ test } from '@jest/globals';
import App from '../src/App';
import { render } from '@testing-library/react';

test.skip('NameConsumer shows default value', () => {
    const AppPage = render(<App />);
    expect(AppPage).toBeDefined();
})

export { }