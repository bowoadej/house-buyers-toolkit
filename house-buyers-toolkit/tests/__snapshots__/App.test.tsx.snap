// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="body"
>
  <div
    className="header"
  >
    <h1>
      House Buyers Toolkit
    </h1>
    <p>
      NOTE:This information is purely for informative purposes only. It does not consitute as financial information and does not consistiute a contract. 
      <br />
       Please contact a certified financial professional for financial advice.
    </p>
  </div>
  <br />
  <div
    className="content"
  >
    <div
      className="accordion-viewing-checklist"
    >
      <div
        className="accordion-item-viewing-checklist"
        id="accordion-item-viewing-checklist"
      >
        <div
          className="accordion-title"
          onClick={[Function]}
        >
          <h2>
            House Viewing Checklist
            <div
              className="accordion-icon"
            >
              +
            </div>
          </h2>
        </div>
      </div>
    </div>
    <div
      className="accordion-maximum-borrowing-amount"
      data-testid="maximum-amount-container"
    >
      <div
        className="accordion-item-maximum-borrowing-amount"
      >
        <div
          className="accordion-title"
          data-testid="accordionTitle"
          onClick={[Function]}
        >
          <h2>
            Maximum Borrowing Amount 
            <div
              className="accordion-icon"
            >
              +
            </div>
          </h2>
        </div>
      </div>
    </div>
    <div
      className="accordion-monthly-repayment-calculator"
    >
      <div
        className="monthly-repayment-accordion-title"
        onClick={[Function]}
      >
        <h2>
          Monthly Repayment Calculator
          <div
            className="accordion-icon"
          >
            +
          </div>
        </h2>
      </div>
    </div>
    <br />
    <div
      className="footer"
    >
      A Toolkit Created By 
      <a
        href="https://bowoadej.com"
      >
        Bowo Adejuyigbe
      </a>
    </div>
    <br />
    <p>
      NOTE:This information is purely for informative purposes only. It does not consitute as financial information and does not consistiute a contract. 
      <br />
       Please contact a certified financial professional for financial advice.
    </p>
  </div>
</div>
`;
